terraform/modules/eks/main.tf

```diff
@@ -0,0 +1,169 @@
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+}
+
+module "eks" {
+  source          = "terraform-aws-modules/eks/aws"
+  version         = "18.10.1"
+  cluster_name    = var.cluster_name
+  cluster_version = "1.21"
+
+  subnet_ids = var.subnet_ids
+  vpc_id     = var.vpc_id
+
+  self_managed_node_group_defaults = {
+    instance_type                = "t2.small"
+  }
+
+  self_managed_node_groups = {
+    worker-group-1 = {
```

Comment:
Here i would suggest to pass an array of self_managed_node_groups where we can define sizes / instances types etc, iterate it and create them instead of hardcoded worker-group-1

Reply:
Same applies here. This is not a public reusable module. I would suggest to keep it simple and leave everything that is common between v1,v2 and v3 to be defined in place and not through variables